### PR TITLE
Switch to Libtask.TapedTask

### PR DESCRIPTION
**Pull request by rikhuijzer**
In response to https://github.com/TuringLang/Libtask.jl/pull/121#issuecomment-1046114428:

> Would be good to evaluate if the alias is needed

Turing and AdvancedPS still rely on CTask so it is still needed currently. This PR will move to the new name.

### CODE REVIEW

**Review by devmotion — APPROVED** [review]

**Comment by devmotion** [comment]
Maybe you can bump the version in the PR?

**Comment by rikhuijzer** [comment]
Done. Can you merge this once CI is done? I don't have merge rights here.

File: Project.toml
Comment (devmotion) [inline Project.toml:51]: This doesn't have to (and probably shouldn't) be updated in this PR:
```suggestion
```

Comment (rikhuijzer) [reply Project.toml]: Okay, I just reverted that commit

**Comment by rikhuijzer** [comment]
Can you merge once CI is done? I don't have rights here

**Review by devmotion — APPROVED** [review]

**Comment by devmotion** [comment]
Sure, I'll merge when tests pass.

File: docs/src/using-turing/advanced.md
Comment (yebai) [inline docs/src/using-turing/advanced.md:194]: It looks good. By the way, soon we can also remove the `TArray` related docs since the new `Libtask` can deep-copy heap-allocated objects - we still need to activate it. See https://github.com/TuringLang/Libtask.jl/blob/4a85b47756259c2778c4336c69b7abc0b61a0fb9/src/tapedfunction.jl#L309

**Review by yebai — APPROVED** [review]

**Pull request merged**
